pkg/transformers/neosync/neosync_email_transformer.go

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package neosync
+
+import (
+	"errors"
+	"fmt"
+	"slices"
+
+	neosynctransformers "github.com/nucleuscloud/neosync/worker/pkg/benthos/transformers"
+	"github.com/xataio/pgstream/pkg/transformers"
+)
+
+type EmailTransformer struct {
+	*transformer[string]
+}
+
+var (
+	errInvalidEmailType          = errors.New("email_type must be one of 'uuidv4', 'fullname' or 'any'")
+	errInvalidInvalidEmailAction = errors.New("invalid_email_action must be one of 'reject', 'passthrough', 'null' or 'generate'")
+
+	validEmailTypes = []string{
+		neosynctransformers.GenerateEmailType_UuidV4.String(),
+		neosynctransformers.GenerateEmailType_FullName.String(),
+		neosynctransformers.GenerateEmailType_Any.String(),
+	}
+	validInvalidEmailActions = []string{
+		neosynctransformers.InvalidEmailAction_Reject.String(),
+		neosynctransformers.InvalidEmailAction_Passthrough.String(),
+		neosynctransformers.InvalidEmailAction_Null.String(),
+		neosynctransformers.InvalidEmailAction_Generate.String(),
+	}
+)
+
+func NewEmailTransformer(params transformers.Parameters) (*EmailTransformer, error) {
+	preserveLength, err := findParameter[bool](params, "preserve_length")
+	if err != nil {
+		return nil, fmt.Errorf("neosync_email: preserve_length must be a boolean: %w", err)
+	}
+
+	preserveDomain, err := findParameter[bool](params, "preserve_domain")
+	if err != nil {
+		return nil, fmt.Errorf("neosync_email: preserve_domain must be a boolean: %w", err)
+	}
+
+	excludedDomains, err := findParameter[[]string](params, "excluded_domains")
+	if err != nil {
+		return nil, fmt.Errorf("neosync_email: excluded_domains must be type of []string: %w", err)
+	}
+	var excludedDomainsAny any
```

Comment:
Right now the value of `excludedDomainsAny` will never be a nil pointer, since you're passing a reference to a variable (a pointer to a nil value vs a nil pointer). You need to add a check so that the value passed to neosync is nil when the param is not found. It might be worth keeping it in a separate function (similar to the `toInt64Ptr` function), it makes it more readable, and it might come in handy later on.

```go 

func toAnyPtr(strArray *[]string) *any {
	if strArray == nil {
		return nil
	}

	var strArrayAny any
	strArrayAny = *strArray
	return &strArrayAny
}

```